src/qutip_cupy/__init__.py

```diff
@@ -58,6 +58,12 @@
 data.trace.add_specialisations([(CuPyDense, cdf.trace_cupydense)])
 data.reshape.add_specialisations([(CuPyDense, CuPyDense, cdf.reshape_cupydense)])
 
+# the whole norm module is imported in QuTiP
+# so the dispatching is peculiar
+data.norm.l2.add_specialisations([(CuPyDense, cdf.l2_cupydense)])
+data.norm.frobenius.add_specialisations([(CuPyDense, cdf.frobenius_cupydense)])
+data.norm.max.add_specialisations([(CuPyDense, cdf.max_cupydense)])
+data.norm.one.add_specialisations([(CuPyDense, cdf.one_cupydense)])
```

Comment:
```suggestion
data.norm.l2.add_specialisations([(CuPyDense, cdf.l2_cupydense)])
data.norm.frobenius.add_specialisations([(CuPyDense, cdf.frobenius_cupydense)])
data.norm.max.add_specialisations([(CuPyDense, cdf.max_cupydense)])
data.norm.one.add_specialisations([(CuPyDense, cdf.one_cupydense)])

```
Putting these into a separate paragraph and removing the comment about the dispatching being weird. This is not weird -- namespacing is a core Python feature!